From 9a82f1c94af88a697c61d31bfb0be6cceae44618 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 12 Aug 2020 12:50:37 +0300
Subject: [PATCH] [SW-17415] gaudi: remove ltssm_en write

It was decided (designers and validation) that Gaudi's LTSSM_EN should
stay at value 0x3 (In Detect and Mask) and not overridden to 0x1 (In
Detect).
Remove the write to LTSSM_EN to keep it at previous value.
We do this in a device specific manner, so other devices will be kept
intact.

Change-Id: I6c7a143f634168b96b29e5cca32260671d9eb072
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/cmd_module.h b/common/cmd_module.h
index bfb88f3..b4eb64f 100644
--- a/common/cmd_module.h
+++ b/common/cmd_module.h
@@ -69,6 +69,7 @@
 void module_pci_teardown(void);
 void pci_dev_specific_setup(void);
 void pci_dev_specific_teardown(void);
+void pci_dev_specific_hot_rst(void);
 
 /* FWUPD */
 void module_fwupd_setup(void);
diff --git a/gaudi/common.c b/gaudi/common.c
index d22e162..a6e5559 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -303,6 +303,11 @@
 	regmap_writel(val, mmPCIE_AUX_FLR_CTRL);
 }
 
+void pci_dev_specific_hot_rst(void)
+{
+	regmap_writel(0x10, mmPCIE_CORE_HOT_RST);
+}
+
 u32 hl_pll_timestamp_base_get(void)
 {
 	return mmCPU_TIMESTAMP_CNTCR;
diff --git a/preboot/module_pci.c b/preboot/module_pci.c
index bf9463a..7f9acbc 100644
--- a/preboot/module_pci.c
+++ b/preboot/module_pci.c
@@ -25,6 +25,12 @@
 __weak void pci_dev_specific_setup(void) {}
 __weak void pci_dev_specific_teardown(void) {}
 
+__weak void pci_dev_specific_hot_rst(void)
+{
+	regmap_writel(0x10, mmPCIE_CORE_HOT_RST);
+	regmap_writel(1, mmPCIE_AUX_LTSSM_EN);
+}
+
 void module_pci_setup(void)
 {
 	if (!hl_is_debug_mode(HL_DEBUG_HW_PCIE))
@@ -60,8 +66,7 @@
 	}
 
 	if (pci_sts.hot_rst_reg_val == 1) {
-		regmap_writel(0x10, mmPCIE_CORE_HOT_RST);
-		regmap_writel(1, mmPCIE_AUX_LTSSM_EN);
+		pci_dev_specific_hot_rst();
 		preboot_puts("HOT RST received\n");
 	}
 }
